frontend/src/component/feature/FeatureView/FeatureVariants/FeatureEnvironmentVariants/PushVariantsButton/PushVariantsButton.tsx

```diff
@@ -0,0 +1,157 @@
+import {
+    Button,
+    Checkbox,
+    Divider,
+    FormControlLabel,
+    Menu,
+    MenuItem,
+    styled,
+} from '@mui/material';
+import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
+import { IFeatureEnvironment } from 'interfaces/featureToggle';
+import { useState } from 'react';
+import { useCheckProjectAccess } from 'hooks/useHasAccess';
+
+const StyledMenu = styled(Menu)(({ theme }) => ({
+    '&>div>ul': {
+        display: 'flex',
+        flexDirection: 'column',
+        justifyContent: 'center',
+        padding: theme.spacing(2),
+        '&>li': {
+            padding: theme.spacing(0),
+        },
+    },
+}));
```

Comment:
I was thinking and if we want to make it perfect maybe we shouldn't remove the padding from the `li`s. It looks very good overall but the hover effect on the items looks a bit weird. Maybe we should remove some of these specifics, clean some of this up, and try to only style a `div` for the separator and button below, but I'm not sure.

Something we can run by @NicolaeUnleash and see what he thinks.

Reply:
Yeah, good point. 
We are actually moving away from `.styles.ts` files and preferring styled components like you're doing here instead, so this looks good as is.